File: src/main/java/unl/cse/Statistics.java
package unl.cse;

/**
 * This is a collection of utilities to compute various statistics on arrays of
 * integers.
 * 
 * @author cbourke
 *
 */
public class Statistics {

	/**
	 * Computes the sum of the given elements.
	 * 
	 * @param array
	 * @return
	 */
	public static int getSum(int array[]) {
		if (array == null || array.length == 0)
			return 0;
		int sum = array[0];
		for (int i = 1; i < array.length; i++) {
			sum = sum + array[i];
		}
		return sum;
	}

	/**
	 * Computes the mean (average) of the given elements.
	 * 
	 * @param array
	 * @return
	 */
	public static double getAverage(int array[]) {
		if (array == null || array.length < 1)
			return 0;
		int sum = getSum(array);
		return sum / (double) array.length;
	}

	/**
	 * Computes the minimal element in the given array of
	 * integers.
	 * 
	 * @param array
	 * @return
	 */
	public static int getMin(int array[]) {
		if (array == null || array.length == 0)
			return 0;
		int min = array[0];
		for (int i = 1; i < array.length; i++) {
			if (array[i] < min) {
				min = array[i];
			}
		}
		return min;
	}

	/**
	 * Computes the maximum element in the given array of
	 * integers.
	 * 
	 * @param array
	 * @return
	 */
	public static int getMax(int array[]) {
		if (array == null || array.length == 0)
			return 0;
		int max = array[0];
		for (int i = 1; i < array.length; i++) {
			if (array[i] > max) {
				max = array[i];
			}
		}
		return max;
	}
}

File: python/statistics.py
"""
A collection statistics-related functions that operate on
lists of numbers.

Note: we've named our functions minimum(), maximum(), and sumVals()
because python already has min(), max(), and sum() functions that
are built-in.  In general, those functions should be used instead of
these (we've only done this to demonstrate basic functionality and
unit testing).
"""
import math

def minimum(values):
    """Finds and returns the minimal element in the given list
    of values.
    """
    minVal = math.inf
    for x in values:
        if x < minVal:
            minVal = x
    return minVal

def maximum(values):
    """TODO: write your own doc-string documentation here
    """
    # TODO: implement
    return

def sumVals(values):
    """TODO: write your own doc-string documentation here
    """
    # TODO: implement
    return

def mean(values):
    """Computes the mean value of the numbers in the given list
    of values.
    """
    return sum(values) / len(values)